velox/docs/functions/spark/array.rst

```diff
@@ -104,13 +104,33 @@ Array Functions
 .. spark:function:: array_sort(array(E)) -> array(E)
 
     Returns an array which has the sorted order of the input array(E). The elements of array(E) must
-    be orderable. Null elements will be placed at the end of the returned array. ::
+    be orderable. Null/NaN elements will be placed at the end of the returned array. ::
 
         SELECT array_sort(array(1, 2, 3)); -- [1, 2, 3]
         SELECT array_sort(array(3, 2, 1)); -- [1, 2, 3]
         SELECT array_sort(array(2, 1, NULL); -- [1, 2, NULL]
         SELECT array_sort(array(NULL, 1, NULL)); -- [1, NULL, NULL]
         SELECT array_sort(array(NULL, 2, 1)); -- [1, 2, NULL]
+        SELECT array_sort(array(4.0, float('nan'), 3.0)); -- [3.0, 4.0, NaN]
+
+.. spark:function:: array_sort(array(T), function(T,U)) -> array(T)
+    :noindex:
+
+    Returns the array sorted by values computed using specified lambda in ascending
+    order. ``U`` must be an orderable type. If the value from the lambda function is NULL, the element will be placed at the end.  ::
```

Comment:
Would you document the behavior for nested nulls? Perhaps also add examples & tests if not added.

Presto array_sort documentation: https://github.com/facebookincubator/velox/blob/main/velox/docs/functions/presto/array.rst?plain=1#L175-L200